src/queue.ts

```diff
@@ -167,18 +167,18 @@ export class Queue<T> {
       success: this.success,
       errored: this.errored,
       retried: this.retried,
-      total: this.tasks.length,
+      total: this.total,
       elapsed: Date.now() - this.startTime,
     };
   }
 
   public taskName(cursorIndex: number): string {
-    const task = this.tasks[cursorIndex];
+    const task = this.tasks[cursorIndex] || "finished task";
     return typeof task === "string" ? task : `index ${cursorIndex}`;
```

Comment:
Could this be simplified to:

```typescript
return this.tasks[cursorIndex] ? `index ${cursorIndex}` : "finished task";
```

Reply:
not quite, because if it's the task itself is a string then the taskName is the string of the task